_posts/2017/2017-11-12-use-a-pull-request-template.markdown

```diff
@@ -0,0 +1,65 @@
+---
+title: 'Use a pull request template'
+description: 'It helps you answer what, why & how.'
+layout: article
+category: ruby
+image:
+  base: '2017/use-a-pull-request-template'
+  alt: 'Man pulling rope'
+  credit: 'Stijn Swinnen'
+  source: 'https://unsplash.com/photos/Q8FHN3qSq2w'
+---
+
+If you're submitting work for review, pull requests are a great way to group related changes and discuss approaches and improvements. The audience for a pull request can be your immediate co-workers, other interested parties or even your future self.
+
+It should describe the ‘what’, ‘why’ and high-level ‘how’ of the changes in your branch. It is important the description is detailed and sets the context for the changes you have made .
+
+Github offers the ability to define pull request templates. See [GitHub’s documentation on pull request templates](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/).
+
+## Instead of…
+
+…writing one line. Or listing a bunch of commits. Or making it up each time.
+
+## Use…
+
+…a template.
+
+### Add `.github/PULL_REQUEST_TEMPLATE.md`
+
+```md
+#### What does this PR do?
+
+[TrelloCard/Issue/Story](LINK_TO_STORY)
+
+##### Why are we doing this? Any context or related work?
+
+#### A good place to start?
+
+#### Manual testing steps?
+
+#### Screenshots
+
+---
+
+#### Migrations
+
+#### Deployment instructions
+
+#### New ENV variables
+```
+
+Each time anyone opens a pull request the description box is pre-filled with the markdown from this file.
+
+
+## But why?
+
+Anything you do over and over works more efficiently if you have a repeatable process to systemise it. Having a template is a solid way to help you write useful pull request descriptions, it helps you pause and answer the questions posed by the default headings in the template.
```

Comment:
There's a lot going on in this sentence. I think you can lose some of the words...

Maybe : "Anything you do over and over works more efficiently if you have an easily repeatable process for it."?

I think the use of 'over and over' _and_ 'efficiently' _and_ 'process' _and_ 'systemise' is a lot!

Reply:
Full-stop after 'descriptions', not a comma.